controllers/usersignup/usersignup_controller.go

```diff
@@ -840,6 +846,12 @@ func (r *Reconciler) ensureSpace(logger logr.Logger, userSignup *toolchainv1alph
 	return space, true, nil
 }
 
+func (r *Reconciler) isSpaceReady(logger logr.Logger, space *toolchainv1alpha1.Space) bool {
+	logger.Info("Ensuring Space is Ready", "Space", space.Name)
+	readyCond, ok := condition.FindConditionByType(space.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	return ok && readyCond.Status == corev1.ConditionTrue && readyCond.Reason == toolchainv1alpha1.SpaceProvisionedReason
```

Comment:
you can use the `IsTrue` function https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/condition/condition.go#L63

Reply:
I want to check the reason as well, ended up using `IsTrueWithReason` https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/condition/condition.go#L91
Thanks for pointing it out

Reply:
done in [5ea85c0](https://github.com/codeready-toolchain/host-operator/pull/670/commits/5ea85c0b2ec4c72293ee59beea0f8b8d3c8af09d)